server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -142,7 +144,7 @@ public HypervisorResource(ConsumerResource consumerResource, ConsumerCurator con
     @UpdateConsumerCheckIn
     @SuppressWarnings("checkstyle:indentation")
     public HypervisorUpdateResultDTO hypervisorUpdate(
-        Map<String, List<GuestIdDTO>> hostGuestDTOMap, @Context Principal principal,
+        Map<String, List<String>> hostGuestMap, @Context Principal principal,
```

Comment:
This change is altering our REST API. In this case, this is the legacy (non-async) hypervisor check-in, which is what legacy virt-who clients are probably using. I'm not certain that this will break them, but I'd be careful about it. From what I understand, the old String argument constructor was used for this legacy use case, where a guestId is sent as string, but there might be another use case where a full-fledged GuestIdDTO is sent. @wottop would be the best person to ask about it.

If this is something we shouldn't change, then we'd have to find a way to make this work otherwise, like this: https://stackoverflow.com/questions/45768194/how-to-generate-constructors-in-swagger-codegen

Reply:
In API call, 'hypervisorUpdate()' method converts the List of guestIds to List of GuestIdDTO using string argument constructor. The rspecs in `hypervisor_check_in.rb` are passing a List of guestIds in string format. So I am assuming, virt-who is using the same format, and this change made successful execution of rspecs. 
@wottop Can you confirm if GuestIdDTO is directly sent?